nevergrad/optimization/differentialevolution.py

```diff
@@ -163,76 +167,43 @@ def _internal_tell(self, x: ArrayLike, value: float) -> None:
-
-    def __init__(self, dimension: int, budget: Optional[int] = None, num_workers: int = 1) -> None:
-        super().__init__(dimension, budget=budget, num_workers=num_workers)
-        self.scale = 1. / np.sqrt(dimension)
-
-
-@base.registry.register
-class MiniQrDE(QrDE):
-
-    def __init__(self, dimension: int, budget: Optional[int] = None, num_workers: int = 1) -> None:
-        super().__init__(dimension, budget=budget, num_workers=num_workers)
-        self.scale = 1. / np.sqrt(dimension)
-
-
-@base.registry.register
-class AlmostRotationInvariantDE(DE):
-
-    def __init__(self, dimension: int, budget: Optional[int] = None, num_workers: int = 1) -> None:
-        super().__init__(dimension, budget=budget, num_workers=num_workers)
-        self.CR = 0.9
+class DifferentialEvolution(base.OptimizerFamily):
+
+    # pylint: disable=unused-argument,too-many-arguments
+    def __init__(self, lhs_init: bool = False, qr_init: bool = False, por_DE: bool = False, scale: Union[str, float] = 1.,
+                 inoculation: bool = False, hyperinoc: bool = False, recommendation: str = "optimistic",
+                 CR: float = .5, F1: float = .8, F2: float = .8, crossover: int = 0):
+        super().__init__(**{x: y for x, y in locals().items() if x not in ["self", "__class__"]})
+        assert recommendation in ["optimistic", "pessimistic", "noisy", "mean"]
+
+    def __call__(self, dimension: int, budget: Optional[int] = None, num_workers: int = 1) -> DE:
+        run = DE(dimension=dimension, budget=budget, num_workers=num_workers)
+        # ugly but effective :s
+        for name, value in self._kwargs.items():
+            rename = name
+            if hasattr(run, "_" + name):
+                rename = "_" + rename
+            elif name == "crossover":
+                rename = "k"
+            elif name == "scale" and isinstance(value, str):
+                if value == "mini":
+                    value = 1. / np.sqrt(dimension)
+                else:
+                    raise ValueError(f'Unknown scaling: "{value}".')
+            setattr(run, rename, value)
+        run.name = repr(self)
+        return run
+
+
+OnePointDE = DifferentialEvolution(crossover=1).with_name("OnePointDE", register=True)
+TwoPointsDE = DifferentialEvolution(crossover=2).with_name("TwoPointsDE", register=True)
+LhsDE = DifferentialEvolution(lhs_init=True).with_name("LhsDE", register=True)
+QrDE = DifferentialEvolution(qr_init=True).with_name("QrDE", register=True)
+MiniDE = DifferentialEvolution(scale="mini").with_name("MiniDE", register=True)
+MiniLhsDE = DifferentialEvolution(lhs_init=True, scale="mini").with_name("MiniLhsDE", register=True)
+MiniQrDE = DifferentialEvolution(qr_init=True, scale="mini").with_name("MiniQrDE", register=True)
+OnePointDE = DifferentialEvolution(recommendation="noisy").with_name("NoisyDE", register=True)
+AlmostRotationInvariantDE = DifferentialEvolution(CR=.9).with_name("AlmostRotationInvariantDE", register=True)
```

Comment:
maybe you can plot the illcond* before and after for checking that it's not worst ?

looks clean to me :-)

Reply:
no visible change + recommendation record is unchanged for tests